components/Observer.vue

```diff
@@ -0,0 +1,32 @@
+<template>
+  <div class="observer" />
+</template>
+
+<script>
+export default {
+  props: {
+    options: {
+      type: Object,
+      default: null
+    }
+  },
+  data: () => ({
+    observer: null
+  }),
+  mounted() {
+    const options = this.options || {}
+    // Use array destructing on the entry argument i.e. entries[0]
+    this.observer = new IntersectionObserver(([entry]) => {
+      if (entry && entry.isIntersecting) {
+        this.$emit('intersect')
+      }
+    }, options)
+
+    // this.$el is the root element of the component
+    this.observer.observe(this.$el)
```

Comment:
Do we know that this works if the element is created off-DOM initially and later added?

Reply:
I can't think why it shouldn't but we would have to test it.  What exactly do you mean by "created off-DOM"?  Dynamically generated in JS and then appended to the DOM?

Reply:
re. "You can create and render a Vue component and then attach it to the actual DOM later. We do that in the map marker code, I think."

The code runs on the mounted method and it's part of Vue so I'm pretty sure it would still work.  Easy enough to test though.